src/npctalk_funcs.cpp

```diff
@@ -741,6 +742,27 @@ void talk_function::buy_100_logs( npc &p )
     add_msg( m_good, _( "%s drops the logs off in the garage…" ), p.get_name() );
 }
 
+/*
+ * Function to make the npc drop non favorite, worn or wielded items at their current position.
+ */
+void talk_function::drop_items_in_place( npc &p )
+{
+    std::vector<drop_or_stash_item_info> to_drop;
+
+    // add all non favorite carried items to the drop off list
+    for( const item_location &npcs_item : p.all_items_loc() ) {
+        if( npcs_item.where() != item_location::type::character && !npcs_item->is_favorite ) {
+            to_drop.emplace_back( npcs_item, npcs_item->count() );
+        }
+    }
+    if( !to_drop.empty() ) {
+        // spawn a activity for the npc to drop the specified items
+        p.assign_activity( player_activity( drop_activity_actor(
+                                                to_drop, tripoint_zero, false
+                                            ) ) );
+    }
```

Comment:
One more thing. If there's nothing to drop, please make the NPC say so.

Reply:
adding both a confirmation and a message if there was nothing to drop.